Week3/homework/step2-6.js

```diff
@@ -2,15 +2,19 @@
 
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
-
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const arr2 = arr.reduce(function(newArr, arrItem) {
+    return newArr.concat(arrItem);
+  }, []);
+  return arr2;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return arr.reduce(
+    (acc, arrItem) =>
+      Array.isArray(arrItem) ? acc.concat(flattenArray3d(arrItem)) : acc.concat(arrItem),
```

Comment:
Very smart to be checking if it's an Array first, and by using a ternary operator. Great work.